kt-code-generators/src/main/kotlin/io/vrap/codegen/kt/languages/php/model/PhpMethodRenderer.kt

```diff
@@ -0,0 +1,180 @@
+        //                            }
+        //                            return "!is_null($body) ? json_encode($body) : null";
+        val content = """
+            |<?php
+            |${PhpSubTemplates.generatorInfo}
+            |namespace ${vrapType.`package`.toNamespaceName().escapeAll()};
+            |
+            |use ${packagePrefix.toNamespaceName().escapeAll()}\\Client\\ApiRequest;
+            |
+            |class ${type.toRequestName()} extends ApiRequest
+            |{
+            |   // const RESULT_TYPE = <returnType(request.returnType)>::class;
+            |
+            |    /**
+            |     <<${type.allParams()?.asSequence()?.map { "* @param ${'$'}$it" }?.joinToString(separator = "\n") ?: "*"}>>
+            |     * @param $!body
+            |     * @param array $!headers
+            |     */
+            |    public function __construct(${type.allParams()?.asSequence()?.map { "$$it, "  }?.joinToString(separator = "") ?: ""}$!body = null, array $!headers = [])
+            |    {
+            |        $!uri = str_replace([${type.allParams()?.asSequence()?.map { "'{$it}'"  }?.joinToString(separator = ", ") ?: ""}], [${type.allParams()?.asSequence()?.map { "$$it"  }?.joinToString(separator = ", ") ?: ""}], '${type.resource().fullUri.template}');
+            |        // <request.method; format="ensureHeader">
+            |        parent::__construct('${type.methodName.toUpperCase()}', $!uri, $!headers, ${type.firstBody()?.serialize()?: "!is_null(\$body) ? json_encode(\$body) : null"});
+            |    }
+            |
+            |    /**
+            |     * @param ResponseInterface $!response
+            |     * @param ResultMapper $!mapper
+            |     * @return <returnType(request.returnType)>
+            |     */
+            |   // public function map(ResponseInterface $!response, ResultMapper $!mapper):  <returnType(request.returnType)>
+            |   // {
+            |   //     return parent::map($!response, $!mapper);
+            |   // }
+            |
+            |   // <if(request.method.queryParameters)>
+            |   // <request.method.queryParameters: {param |<withParam(request.name, param)>}>
+            |   // <endif>
+            |}
+        """.trimMargin().keepIndentation("<<", ">>").forcedLiteralEscape()
+        val relativeTypeNamespace = vrapType.`package`.toNamespaceName().replace(packagePrefix.toNamespaceName() + "\\", "")
+        val relativePath = "src/" + relativeTypeNamespace.toNamespaceDir() + "/" + type.toRequestName() + ".php"
+        return TemplateFile(
+                relativePath = relativePath,
+                content = content
+        )
+    }
+
+    private fun Body.serialize(): String {
+        if (this.type is FileType) {
+            return "!is_null(\$body) ? \$body->getStream() : null"
+        }
+        return "!is_null(\$body) ? json_encode(\$body) : null"
+    }
+
+
+
+
+
+//    fun getAllParamNames(): List<String>? {
```

Comment:
just to avoid the later cleanup maybe we can review the comment part straight from here?